xla/service/cpu/onednn_ops_rewriter.cc

```diff
@@ -95,24 +121,25 @@ std::optional<HloInstruction*> MatchSoftmax(HloInstruction* instr) {
   HloInstruction* left_exponential;
   HloInstruction* right_exponential;
   HloInstruction* left_producer;
-  HloInstruction* right_producer;
+  HloInstruction* reduce_sum;
+  HloInstruction* reduce_max;
+  HloInstruction* reduce_instr;
 
   // Lower diamond
   if (!Match(instr,
              m::Divide(
                  m::Exp(&left_exponential, m::Op()),
                  m::Broadcast(m::Reshape(
                      m::Broadcast(OptionalConvert(m::Reshape(OptionalConvert(
-                         m::Reduce(OptionalConvert(
+                         m::Reduce(&reduce_sum,
```

Comment:
Why do we match `max(reduce_max, min)` above but not `add(reduce_sum, 0)` here?

Reply:
We added max(reduce_max, -inf) based on the patterns we saw for actual jax models, we didn't encounter a case for later for jax/TF models.

Reply:
Thanks for explaining. Would you mind adding a comment with this note?

Reply:
Got it, thanks for explaining. Would you mind adding a comment in the code to explain that this is necessary for cases you saw in practice?

Reply:
Modified the description of pattern matching to address this